docs: Refactor the installation instructions for `s-nomp`

upbqdn opened the pull request:
## Motivation

When working on #7423, I noticed our documentation for mining with `s-nomp` was outdated.

## Solution

- Don't mention the `getblocktemplate-rpcs` feature anymore.
- Refactor the installation instructions for `s-nomp`. I only did this refactor for the Arch Linux section.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

upbqdn committed:
Remove references to `getblocktemplate-rpcs`

upbqdn committed:
Refactor the sync description

upbqdn committed:
Refactor the instructions for installing `s-nomp`

upbqdn commented on book/src/user/mining-testnet-s-nomp.md:
````diff
@@ -137,43 +138,49 @@ These fixes disable mining pool operator payments and miner payments: they just
+   ```
 
-1. `git clone https://github.com/ZcashFoundation/s-nomp && cd s-nomp`
+5. Use Node 8.11:
````
```suggestion
5. Use Node 10:
```

teor2345 commented on book/src/user/mining-testnet-s-nomp.md:
````diff
@@ -73,11 +73,12 @@ These fixes disable mining pool operator payments and miner payments: they just
     ```
-3. Wait a few hours for Zebra to sync to the testnet tip (on mainnet this takes 2-3 days)
+3. Wait for Zebra to sync to the testnet tip. This takes 2-3 days on Mainnet at
+   the time of writing this manual.
````
I know the testnet sync time if that helps?

Also I usually put dates so we know when things should be updated. (And users know how outdated they are.)

```suggestion
3. Wait for Zebra to sync to the testnet tip. This takes 8-12 hours on testnet (or 2-3 days on mainnet) as of October 2023.
```

upbqdn replied:
Yea, that's much better. Thank you.

upbqdn committed:
Mention Node 10 instead of 8.11

upbqdn committed:
Refactor the description of Zebra's sync

Co-authored-by: teor <teor@riseup.net>

upbqdn commented:
Note that I refactored the install instructions only for the Arch Linux section.

upbqdn commented:
> Note that I refactored the install instructions only for the Arch Linux section.

Also, the general instructions don't work anymore as they miss some necessary steps, such as using Python 3.10. I updated only the Arch section since I tested the new instructions only on that distribution. I thought this is acceptable since we don't plan to rely on `s-nomp` substantially.

teor2345 reviewed (approved):
I think that's fine for now
